pkg/machines/components/vm/vmExpandedContent.jsx

```diff
@@ -17,49 +17,167 @@
-        { name: _("Network interfaces"), id: cockpit.format("$0-networks", vmId(vm.name)), renderer: VmNetworkTab, presence: 'onlyActive', data: { vm, dispatch, config, hostDevices, interfaces, networks, nodeDevices, onAddErrorNotification } },
-        { name: _("Consoles"), id: cockpit.format("$0-consoles", vmId(vm.name)), renderer: Consoles, data: { vm, config, dispatch, onAddErrorNotification } },
-    ];
-    if (vm.snapshots !== -1)
-        tabRenderers.splice(4, 0, { name: _("Snapshots"), id: cockpit.format("$0-snapshots", vmId(vm.name)), renderer: VmSnapshotsTab, data: { vm, dispatch, config, onAddErrorNotification } });
+    useEffect(() => {
+        // Anything in here is fired on component mount.
+        onUsageStartPolling();
+        return () => {
+            // Anything in here is fired on component unmount.
+            onUsageStopPolling();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
 
-    let initiallyActiveTab = null;
-    if (vm.ui && vm.ui.initiallyOpenedConsoleTab) {
-        initiallyActiveTab = tabRenderers.map((o) => o.name).indexOf(_("Consoles"));
+    if (cockpit.location.path[1] == "console") {
+        return (<Page breadcrumb={
+            <Breadcrumb className='machines-listing-breadcrumb'>
+                <BreadcrumbItem to='#'>
+                    {_("Virtual machines")}
+                </BreadcrumbItem>
+                <BreadcrumbItem onClick={() => cockpit.location.go(["vm"], Object.assign(cockpit.location.options, { name: vm.name, connection: vm.connectionName }))}>
+                    <a className="pf-c-breadcrumb__link">{vm.name}</a>
+                </BreadcrumbItem>
+                <BreadcrumbItem isActive>
+                    {_("Console")}
+                </BreadcrumbItem>
+            </Breadcrumb>}>
+            <PageSection variant={PageSectionVariants.light}>
+                <Consoles vm={vm} config={config} dispatch={dispatch}
+                          onAddErrorNotification={onAddErrorNotification} />
+            </PageSection>
+        </Page>);
     }
 
-    return (vm.snapshots !== undefined
-        ? <ListingPanel
-            colSpan='4'
-            initiallyActiveTab={initiallyActiveTab}
-            tabRenderers={tabRenderers} />
-        : null);
+    const cardContents = [
+        {
+            id: `${vmId(vm.name)}-overview`,
+            title: _("Overview"),
+            body: <VmOverviewTab vm={vm} config={config} dispatch={dispatch}
+                                 nodeDevices={nodeDevices} libvirtVersion={libvirtVersion} />,
+        },
+        {
+            id: `${vmId(vm.name)}-usage`,
+            className: 'usage-card',
+            title: _("Usage"),
+            body: <VmUsageTab vm={vm} />,
+        },
+        {
+            id: `${vmId(vm.name)}-consoles`,
+            className: "consoles-card",
+            title: <>{_("Console")}<Button variant="link"
```

Comment:
This looks like only the icon would be clickable -- shouldn't the "Console" word be part of the link and get clickable as well?

Reply:
We should have text next to the icon that says "maximize" or "expand".

It's already mentioned in the follow-up PR @ https://github.com/cockpit-project/cockpit/issues/14604 — I'm waiting for this PR to land to address those issues.